modules/core/include/opencv2/core/ovx_defs.hpp

```diff
@@ -0,0 +1,42 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+// Copyright (C) 2016, Intel Corporation, all rights reserved.
+// Third party copyrights are property of their respective owners.
+
+// OpenVX related definitions and declarations
+
+#pragma once
+#ifndef OPENCV_OVX_DEFS_HPP
+#define OPENCV_OVX_DEFS_HPP
+
+#include "cvdef.h"
+#include "cvconfig.h"
+
+namespace cv
+{
+/// Check if use of OpenVX is possible
+CV_EXPORTS_W bool haveOpenVX();
+
+/// Check if use of OpenVX is enabled
+CV_EXPORTS_W bool useOpenVX();
+
+/// Enable/disable use of OpenVX
+CV_EXPORTS_W void setUseOpenVX(bool flag);
+} // namespace cv
+
+// utility macro for running OpenVX-based implementations
+#ifdef HAVE_OPENVX
+
+#define CV_OVX_RUN(condition, func, ...)          \
```

Comment:
This is not related to public interface. You need to slit this file into public API and private OpenCV-only API.